### ISSUE TITLE
Gauntlet 2: Parametric Floor — Why Keep Data?

### ISSUE THREAD

**Issue by samscarrow** [git-scarrow/fast-associative-memory#2]
# Gauntlet 2 — "Parametric Floor" Test (Why Keep Data?)

## Purpose
Red Team Phase 1. Determine whether a single linear layer trained on the streaming data matches or exceeds FAM accuracy. If it does, the entire associative memory mechanism (storing instances) is unnecessary overhead.

## Kill Condition
If a linear probe achieves ≥ FAM top-1 accuracy, the concept of Associative Memory is unnecessary overhead.

## Protocol
- **Backbone:** Same frozen DINOv2 as Gauntlet 1. No fine-tuning.
- **Model:** Single linear layer `(D × num_classes)`. No hidden layers.
- **Dataset:** Same streaming eval split as Gauntlet 1. Same class/batch order.
- **Report:** top-1 accuracy and wall-clock training time (seconds).

### 2a — Streaming SGD / AdamW Linear Probe
Standard online linear probe with a small replay buffer.

```python
# Linear layer: nn.Linear(D, num_classes, bias=True)
# Optimizer: AdamW, lr=1e-3
# Replay buffer: ring buffer, capacity=256 samples, random replacement
# Each step: sample mini-batch from buffer + current batch, gradient step
# Buffer constraint: O(buffer_size * D) memory — must be << FAM memory budget
```

### 2b — Streaming Ridge Regression (Woodbury / Incremental Closed-Form)
Closed-form solution updated incrementally using the Woodbury identity (Sherman-Morrison rank-1 update). No replay buffer required — O(D²) memory for the covariance matrix.

```python
# Maintain: A = X^T X + lambda * I  (D x D), b = X^T y  (D x num_classes)
# Update: rank-1 update to A^{-1} via Sherman-Morrison each new sample
# Predict: W = A^{-1} b, logits = emb @ W
```

## Deliverables
1. `benchmarks/gauntlet_02_linear.py` — standalone script that:
   - Implements both 2a (SGD/AdamW probe) and 2b (Streaming Ridge)
   - Runs on the streaming eval set with frozen DINOv2 embeddings
   - Prints a results table: `method | top-1 accuracy | memory (MB) | train time (s)`
2. Results reproducible with fixed seed.
3. CLI args: `--dataset`, `--backbone`, `--seed`, `--buffer-size`, `--lambda`, `--output`.

## Acceptance
Script runs to completion and prints the results table. FAM accuracy included for direct comparison.
